src/main/java/com/clevertap/stormdb/index/IndexMap.java

```diff
@@ -0,0 +1,38 @@
+package com.clevertap.stormdb.index;
+
+public interface IndexMap {
+
+    /**
+     * Mapping having this as the key should not be allowed put since it represents null.
+     */
+    int NO_ENTRY_KEY = Integer.MAX_VALUE;
+
+    /**
+     * Mapping having this as the value should be returned when get fails or in essence
+     * represents null value index.
+     */
+    int NO_ENTRY_INDEX_VALUE =  0xffffffff;
+
+
+    /**
+     * API to support put for the key/index pair in question. Key {@link IndexMap#NO_ENTRY_KEY}
+     * should be block
+     * @param key The key to be inserted
+     * @param indexValue The index mapping for the key
+     */
+    abstract void put(int key, int indexValue);
```

Comment:
I don't think `abstract` is required here.

Reply:
Good point. Switched from abstract class to interface for better clarity.